Scripts/Proxies/TwoHandedProxyBase.cs

```diff
@@ -161,6 +182,100 @@ public virtual void Update()
 				m_RightHand.localRotation = trackedObjectInput.rightRotation.quaternion;
 			}
 		}
+
+		public void AddFeedbackRequest(FeedbackRequest request)
+		{
+			var proxyRequest = request as ProxyFeedbackRequest;
+			if (proxyRequest != null)
+			{
+				m_FeedbackRequests.Add(proxyRequest);
+				ExecuteFeedback(proxyRequest);
+			}
+		}
+
+		void ExecuteFeedback(ProxyFeedbackRequest changedRequest)
+		{
+			foreach (var proxyNode in m_Buttons)
+			{
+				foreach (var kvp in proxyNode.Value)
+				{
+					ProxyFeedbackRequest request = null;
+					foreach (var req in m_FeedbackRequests)
+					{
+						var matchChanged = req.node == changedRequest.node && req.control == changedRequest.control;
+						var matchButton = req.node == proxyNode.Key && req.control == kvp.Key;
+						var sameCaller = req.caller == changedRequest.caller;
+						var priority = request == null || req.priority >= request.priority;
+						if (matchButton && priority && (matchChanged || sameCaller))
+							request = req;
+					}
+
+					if (request == null)
+						continue;
+
+					var button = kvp.Value;
+					if (button.renderer)
+						this.SetHighlight(button.renderer.gameObject, true, duration: k_FeedbackDuration);
+
+					if (button.transform)
+					{
+						var tooltip = button.transform.GetComponent<Tooltip>();
+						var tooltipText = request.tooltipText;
+						if (!string.IsNullOrEmpty(tooltipText) && tooltip)
+						{
+							tooltip.tooltipText = tooltipText;
+							this.ShowTooltip(tooltip, true, k_FeedbackDuration);
+						}
+					}
+				}
+			}
+		}
+
+		public void RemoveFeedbackRequest(FeedbackRequest request)
+		{
+			var proxyRequest = request as ProxyFeedbackRequest;
+			if (proxyRequest != null)
+				RemoveFeedbackRequest(proxyRequest);
+		}
+
+		void RemoveFeedbackRequest(ProxyFeedbackRequest request)
+		{
+			Dictionary<VRInputDevice.VRControl, ProxyHelper.ButtonObject> buttons;
+			if (m_Buttons.TryGetValue(request.node, out buttons))
+			{
+				ProxyHelper.ButtonObject button;
+				if (buttons.TryGetValue(request.control, out button))
+				{
+					if (button.renderer)
+						this.SetHighlight(button.renderer.gameObject, false);
```

Comment:
Yes. This will unset highlights/tooltips on this button from any source. This is why we call ExecuteFeedback afterward, which will re-execute any underlying feedback which we might have stomped.